src/lib/tasker.ts

```diff
@@ -0,0 +1,171 @@
+import 'dotenv/config';
+import AppDataSource from '../api/data-source.js';
+import { Attempt } from '../api/entities/attempt.js';
+import { Test } from '../api/entities/test.js';
+import { TestResult, TestResultStatus } from '../api/entities/test-result.js';
+import { evaluateCompression } from './evaluate.js';
+
+const POLL_INTERVAL = process.env.TASKER_POLL_INTERVAL
+  ? parseInt(process.env.TASKER_POLL_INTERVAL, 10)
+  : 5000;
+
+/**
+ * Finds all active tests that do not have a result for the given attempt and
+ * processes each of these tests in parallel
+ *
+ * For each test, it will:
+ * a. Create a "pending" test result to lock the test.
+ * b. Run the compression and evaluation logic.
+ * c. Update the test result with the final outcome and metrics.
+ * d. Mark the test as failed if any step fails.
+ *
+ * @param attempt The attempt context, containing the compressing prompt and model information.
+ */
+async function processAttempt(attempt: Attempt): Promise<void> {
+  console.log(`Processing attempt ${attempt.id}`);
+  const testRepo = AppDataSource.getRepository(Test);
+  const testResultRepo = AppDataSource.getRepository(TestResult);
+  // Find all active tests for the current attempt that don't have a result yet.
+  const testsToProcess = await testRepo
+    .createQueryBuilder('test')
+    .leftJoin('test.testResults', 'tr', 'tr.attempt_id = :attemptId', {
+      attemptId: attempt.id,
+    })
+    .where('test.isActive = true')
+    .andWhere('tr.attempt_id IS NULL')
+    .getMany();
+  if (testsToProcess.length === 0) {
+    console.log(`Attempt ${attempt.id} has no more tests to process.`);
+    return;
+  }
+  console.log(
+    `Found ${testsToProcess.length} tests to process for attempt ${attempt.id}`
+  );
+  await Promise.all(
```

Comment:
Process the tests sequentially, not in parallel. we want to cancel processing if one of them fails. In JS, promises are not cancellable, so Promise.all should not be used. just do it in a loop